aspnetcore/signalr/scale.md

```diff
@@ -0,0 +1,87 @@
+---
+title: ASP.NET Core SignalR production hosting and scaling
+author: tdykstra
+description: Learn how to avoid performance and scaling problems in apps that use ASP.NET Core SignalR.
+monikerRange: '>= aspnetcore-2.1'
+ms.author: tdykstra
+ms.custom: mvc
+ms.date: 11/28/2018
+uid: signalr/scale
+---
+
+# ASP.NET Core SignalR hosting and scaling
+
+By [Andrew Stanton-Nurse](https://twitter.com/anurse), [Brady Gaster](https://twitter.com/bradygaster), and [Tom Dykstra](https://github.com/tdykstra),
+
+This article explains hosting and scaling considerations for high-traffic apps that use ASP.NET Core SignalR.
+
+## TCP connection resources
+
+The number of concurrent TCP connections that a web server can support is limited. Standard HTTP clients use *ephemeral* connections. These connections can be closed when the client goes idle and re-opened later. On the other hand, a SignalR connection is *persistent*. SignalR connections stay open even when the client goes idle. In a high-traffic app that serves many clients, these persistent connections can cause servers to hit their maximum number of connections.
+
+Persistent connections also consume some additional memory, to track each connection.
+
+The heavy use of connection-related resources by SignalR can affect other web apps that are hosted on the same server. When SignalR opens and holds the last available TCP connections, other web apps on the same server also have no more connections available to them.
+
+If a server runs out of connections, you'll see random socket errors and connection reset errors, for example:
+
+```
+An attempt was made to access a socket in a way forbidden by its access permissions...
+```
+
+To keep SignalR resource usage from causing errors in other web apps, run SignalR on different servers than your other web apps.
```

Comment:
```suggestion
To prevent SignalR resource consumption from causing errors in other web apps, run SignalR on different servers than your other web apps.
```